agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1221,31 +1222,60 @@ func (dg *dockerGoClient) APIVersion() (dockerclient.DockerVersion, error) {
 	return dg.sdkClientFactory.FindClientAPIVersion(client), nil
 }
 
-// Stats returns a channel of *docker.Stats entries for the container.
-func (dg *dockerGoClient) Stats(id string, ctx context.Context) (<-chan *docker.Stats, error) {
-	client, err := dg.dockerClient()
+// Stats returns a channel of *types.Stats entries for the container.
+func (dg *dockerGoClient) Stats(id string, ctx context.Context) (<-chan *types.Stats, error) {
+	client, err := dg.sdkDockerClient()
 	if err != nil {
 		return nil, err
 	}
 
-	stats := make(chan *docker.Stats)
-	options := docker.StatsOptions{
-		ID:                id,
-		Stats:             stats,
-		Stream:            true,
-		Context:           ctx,
-		InactivityTimeout: StatsInactivityTimeout,
-	}
+	// Create channel to hold the stats
+	statsChnl := make(chan *types.Stats)
+	var resp types.ContainerStats
 
 	go func() {
-		statsErr := client.Stats(options)
-		if statsErr != nil {
-			seelog.Infof("DockerGoClient: Unable to retrieve stats for container %s: %v",
-				id, statsErr)
+		// ContainerStats outputs a io.ReadCloser and an OSType
+		stream := true
+		resp, err = client.ContainerStats(ctx, id, stream)
+		if err != nil {
+			seelog.Warnf("DockerGoClient: Unable to retrieve stats for container %s: %v", id, err)
+			close(statsChnl)
+			return
 		}
+
+		// handle inactivity timeout
+		_, cancelRequest := context.WithCancel(ctx)
+		defer cancelRequest()
+		var canceled uint32
+		var ch chan<- struct{}
+		resp.Body, ch = handleInactivityTimeout(resp.Body, StatsInactivityTimeout, cancelRequest, &canceled)
+		defer close(ch)
+
```

Comment:
resp.Body is not always closed in the following code, probably should add defer res.Body.Close() here?